main/src/com/google/refine/templating/Parser.java

```diff
@@ -86,6 +92,9 @@ static public Template parse(String s) throws ParsingException {
                     String expression = s.substring(current + 2, closeBrace);
 
                     if (current > start) {
+                    	if (!s.substring(start, current).matches("\\s*,?\\s*\\\"*[a-zA-Z0-9() ]*\\\"\\s*:\\s*")) {
```

Comment:
It is not clear to me which case this is catching, could you expand a bit, maybe with a comment?

Reply:
I see, this is to test the keys of the json, i.e, ```"URL"```, ```"LA (name)"```. Will add a comment here :)

Reply:
I am not sure what JSON has to do with the context here - the templating exporter can be used to generate JSON, but it can be used to generate anything else as well, so we should definitely not build in anything specific to JSON in this parser as far as I can tell. But maybe I missed something?

Reply:
I see, by json I mean the template keys. For instance the keys - ```"URL" :``` and ```"LA (name)" :```. Currently, I have added a regex expression such that it would allow only alphabets and parentheses between the double-quotes. My question is should we allow ```"URL{" :``` this, i.e, curly brackets in between the double-quotes. And should we restrict only to alphabets and parentheses?

Reply:
There is nothing special about "URL" or "LA (name)" in the test cases, they are just some substrings in the static parts of the template. There is no reason to treat them specially.
For instance, a template such as `{ "UR{N": {{jsonize(cells["URN"].value)}}, "foo:"bar }` would be totally legitimate. It would not generate valid JSON but the templating exporter does not know anything about the expected format of what it is generating. All it knows is that a template should be a series of static and dynamic fragments. The static fragments can be arbitrary strings (once the braces have been unescaped). The dynamic fragments must be valid expressions but those expressions can generate arbitrary strings too.